File: app/src/test/resources/testcode-cs/ClassUsagePatterns.cs
// Test file to verify C# class usage patterns

using System;
using System.Collections.Generic;

public class BaseClass
{
    public static void StaticMethod()
    {
        Console.WriteLine("static");
    }

    public void InstanceMethod()
    {
        Console.WriteLine("instance");
    }
}

public class ExtendedClass : BaseClass
{
    // Field with type
    private BaseClass field;

    public ExtendedClass(BaseClass param)
    {
        // Constructor call
        this.field = new BaseClass();

        // Static access
        BaseClass.StaticMethod();
    }

    // Method with type annotations
    public BaseClass ProcessBase(BaseClass input)
    {
        return input;
    }

    // Generics
    public List<BaseClass> GetList()
    {
        return new List<BaseClass>();
    }
}

public class UsageExample
{
    // Variable declaration
    private BaseClass myBase = new BaseClass();
}

File: brokk-shared/src/testFixtures/resources/testcode-ts/ClassUsagePatterns.ts
// Test file to verify TypeScript class usage patterns

export class BaseClass {
    static staticMethod(): string {
        return "static";
    }

    instanceMethod(): string {
        return "instance";
    }
}

interface IBase {
    processBase(input: BaseClass): BaseClass;
}

class ExtendedClass extends BaseClass implements IBase {
    // Field with type annotation
    private field: BaseClass;

    constructor(param: BaseClass) {
        super();

        // Constructor call
        this.field = new BaseClass();

        // Static access
        const result = BaseClass.staticMethod();
    }

    // Method with type annotations
    processBase(input: BaseClass): BaseClass {
        return input;
    }

    // Generics
    getList(): Array<BaseClass> {
        return [new BaseClass()];
    }

    // Arrow function with return type
    getter = (): BaseClass => {
        return this.field;
    }
}

// Variable with type annotation
const myBase: BaseClass = new BaseClass();

// Generic array
const list: Array<BaseClass> = [];

// Import/export
export { ExtendedClass };
